src/validators/float.rs

```diff
@@ -108,9 +108,23 @@ impl Validator for ConstrainedFloatValidator {
             return Err(ValError::new(ErrorTypeDefaults::FiniteNumber, input));
         }
         if let Some(multiple_of) = self.multiple_of {
-            let rem = float % multiple_of;
-            let threshold = float.abs() / 1e9;
-            if rem.abs() > threshold && (rem - multiple_of).abs() > threshold {
+            const EPSILON_FACTOR: f64 = 100.0;
+            let epsilon_threshold = f64::EPSILON * EPSILON_FACTOR;
+
+            // Round the result of dividing the input value by the multiple
+            let rounded = (float / multiple_of).round();
+
+            // Calculate the difference between the rounded value and the original value
+            let diff = (float - rounded * multiple_of).abs();
```

Comment:
It feels to me like given we evaluate `threshold` in units of `multiple_of`, we might want to do similarly here to end up with same scale of `multiple_of` (by dividing through by `rounded`)?

```suggestion
            // Calculate how close the rounded division is to multiple_of
            let diff = (float / rounded - multiple_of).abs();
```